### PR TITLE
Improve documentation and examples

### PR DESCRIPTION
**Pull request by gsserge**
Closes https://github.com/rust-vmm/vm-device/issues/28

Signed-off-by: Sergii Glushchenko <gsserge@amazon.com>

### CODE REVIEW

File: src/device_manager.rs [hunk 1]
```diff
@@ -3,12 +3,116 @@
...
+//!
+//! Registering a new device can be done using the register methods of [`PioManager`]
+//! and [`MmioManager`] with an appropriate bus range
+//! ([`PioRange`](../bus/type.PioRange.html) or [`MmioRange`](../bus/type.MmioRange.html)).
+//! ```
+//! # use std::sync::Arc;
+//! # use vm_device::bus::{PioAddress, PioAddressOffset, PioRange};
+//! # use vm_device::bus::{MmioAddress, MmioAddressOffset, MmioRange};
+//! # use vm_device::device_manager::{IoManager, PioManager, MmioManager};
+//! # use vm_device::{DevicePio, DeviceMmio};
```
Comment (andreeaflorescu) [inline src/device_manager.rs:26]: nit: there are lots of imports compared to the number of lines of example code, maybe we can hide them from the example by using ` # ` in front of `use ...`.

**Review by andreeaflorescu — DISMISSED** [review]

File: src/device_manager.rs [hunk 2]
```diff
@@ -3,12 +3,116 @@
...
 //! deregistering devices and helping VM IO exit handling.
-//！VMM would be responsible for getting device resource request, ask
-//! vm_allocator to allocate the resources, ask vm_device to register the
-//! devices IO ranges, and finally set resources to virtual device.
+//! It defines two buses, one for PIO and one for MMIO, and provides default
+//! implementations of [`PioManager`] and [`MmioManager`].
+//!
+//! VMM would be responsible for getting device resource request, allocating the
+//! resources, asking `vm_device` to register the devices IO ranges, and finally
+//! setting resources to virtual device.
```
Comment (lauralt) [inline src/device_manager.rs:14]: I know this phrase wasn't added in this PR, but it sounds a bit strange, e.g. `asking vm_device something`; the `getting device resource request` is also confusing (for me at least). Can we somehow rephrase it please?

File: src/lib.rs [hunk 3]
```diff
@@ -1,7 +1,52 @@
 // Copyright © 2019 Intel Corporation. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
 
-//! rust-vmm device model.
```
Comment (lauralt) [inline src/lib.rs:4]: I think we can also add the `#![deny(missing_docs)]` now. It also looks like [this public method](https://github.com/rust-vmm/vm-device/blob/main/src/bus/range.rs#L44) is not documented yet.

Comment (andreeaflorescu) [reply src/lib.rs:4]: Good catch!

File: src/device_manager.rs (hunk 2 above)
Comment (gsserge) [reply src/device_manager.rs:14]: Please suggest how can we rephrase this.

Comment (andreeaflorescu) [reply src/device_manager.rs:14]: "The VMM must first allocate unique resources (such as bus ranges), and then call into the `vm-device` interface to register the devices with their corresponding resources."
WDYT?

Comment (lauralt) [reply src/device_manager.rs:14]: Sounds good to me, thanks.

File: src/bus/range.rs [hunk 4]
```diff
@@ -81,8 +82,9 @@ impl<A: BusAddress> Ord for BusRange<A> {
     }
 }
 
-// Helper type aliases.
+/// Represents a MMIO bus range.
```
Comment (lauralt) [inline src/bus/range.rs:85]: nit:
```suggestion
/// Represents an MMIO bus range.
```

File: src/bus/range.rs [hunk 5]
```diff
@@ -41,6 +41,7 @@ impl<A: BusAddress> BusRange<A> {
         self.base
     }
 
+    /// Return the number of attached devices.
```
Comment (lauralt) [inline src/bus/range.rs:44]: Isn't this returning the size of the range?

**Review by lauralt — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**